Merging some new files from PX4 forks of RotorS and making some launch and description files more generic

pvechersky opened the pull request:
Merging some new files from PX4 fork of RotorS:
- A couple of new models: ardrone and iris
- A new 'house' world
- A mavlink_interface gazebo plugin
- A px4 dummy controller

Leaving out some of the following from PX4 fork:
- Unfinished wing_model gazebo plugin
- Unfinished implementation of a VTOL model description

In addition to processing the PX4 fork differences I made the launch and description files a little more generic by adding a 'world_name' argument to the launch files and by consolidating the '<mav_name>_generic_odometry_sensor.gazebo' files into one 'mav_generic_odometry_sensor.gazebo' file and same with the wind gust and vi_sensor files.

ethzasl-jenkins commented:
Can one of the admins verify this patch?

ffurrer commented:
add to whitelist

helenol commented on rotors_control/include/rotors_control/px4_dummy_controller.h:
```diff
@@ -0,0 +1,67 @@
+
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+ private:
+	ros::NodeHandle _n;
```
Please follow google style -- 2 spaces rather than tabs everywhere, member variables are named motor_speeds_ (trailing underscore), consistency on which side the * and & operators go (pick one of `Eigen::VectorXd* name` or `Eigen::VectorXd *name` and stick with it throughout), spaces between arguments, etc.

https://google.github.io/styleguide/cppguide.html

helenol commented on rotors_control/src/px4_dummy_controller.cpp:
```diff
@@ -0,0 +1,87 @@
+
+void PX4dummyController::InitializeParams() {
+
+  amount_rotors_ = 4;
```
Set these in the constructor.
Also "amount_rotors_" can NEVER be anything other than 4 as you allocate a fixed-size matrix in the class description. Either change it to a std::vector of the correct size, or don't allow a variable for this.

helenol commented on rotors_control/src/px4_dummy_controller.cpp:
```diff
@@ -0,0 +1,87 @@
+
+void PX4dummyController::MotorVelCallback(const mav_msgs::MotorSpeed &msg) {
+  //XXX sometimes this is called on a uninitialized instance: find out why!
+  InitializeParams();
```
Unnecessary if set in the constructor. Also why have `initialized_params_` if you don't use it?

helenol commented on rotors_gazebo_plugins/package.xml:
```diff
@@ -3,7 +3,7 @@
   <description>The rotors_gazebo_plugins package</description>
 
-  <maintainer email="fadri.furrer@mavt.ethz.ch">Fadri Furrer</maintainer>
+  <maintainer email="markus.achtelik_devel@mavt.ethz.ch">Markus Achtelik</maintainer>
```
Why this change?

helenol commented on rotors_gazebo_plugins/package.xml:
```diff
@@ -24,6 +24,7 @@
   <build_depend>gazebo</build_depend>
   <build_depend>geometry_msgs</build_depend>
   <build_depend>mav_msgs</build_depend>
+  <build_depend>octomap_ros</build_depend>
```
Now it's in twice, please remove duplicates.

helenol commented on rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp:
```diff
@@ -0,0 +1,236 @@
+  gravity_W_ = world_->GetPhysicsEngine()->GetGravity();
+
+  // Magnetic field data for Zurich from WMM2015 (10^5xnanoTesla (N, E, D))
+  mag_W_ = {0.21523, 0.00771, 0.42741};
```
Please set all these params from constants in the header so that it's obvious how to change these/what these are.

burrimi commented on rotors_control/include/rotors_control/px4_dummy_controller.h:
```diff
@@ -0,0 +1,67 @@
+ */
+
+#include <rotors_control/controller_base.h>
+#include <rotors_control/controller_factory.h>
```
We removed the controller factory and controller base class to simplify the code.
This will not compile.

pvechersky commented:
Thank you for the feedback! I've been working on some of these changes, hence been a little slow to respond.

The code segments on which you have commented are from the master branch of PX4's fork of RotorS. Sorry that I included them in the pull request without taking too close of a look first :/

I also realized the problem with the px4_dummy_controller class and also the fact that it is not even fully implemented. Therefore, I will leave it out of this pull request. I will also leave out the portion of gazebo_mavlink_interface that deals with motor speeds as it has dependencies from PX4's fork of Mav-comm and would further complicate things for now.

I will be submitting some corrections shortly. Thank you!

pvechersky commented:
Hi guys, I made some more changes to this pull request:
- With Fadri's help I created parameter YAML files for new models (Ardrone and Iris) so that they can fly
- Modified the mesh file for Ardrone to bring it down to the right scale
- Refactored the mavlink interface plugin the best I could to bring it up to the standards of other plugins
- Restored the functionality of mavlink interface plugin to receive HIL control commands and changed this code to work with currently available mav_msgs type and to talk to the motor model plugin

Thank you!

ffurrer commented:
lgtm @helenol and @burrimi can you also take another look please?

helenol commented:
LGTM, huge improvement thanks!
